src/v2/filter/kuadrant_filter.rs

```diff
@@ -32,6 +34,21 @@ impl Context for KuadrantFilter {
         );
         if let Some(pipeline) = self.pipeline.take() {
             self.pipeline = pipeline.digest(token_id, status_code, response_size);
+
+            if !self.should_pause() {
+                let result = if self.in_response_phase {
+                    self.resume_http_response()
+                } else {
+                    self.resume_http_request()
+                };
+
+                if let Err(e) = result {
+                    error!(
+                        "#{} failed to resume filter processing: {:?}",
+                        self.context_id, e
+                    );
+                }
```

Comment:
We might have to do more than logging only here; I think maybe we should panic/fatal? I _think_ this will deadlock indefinitely